tests/test_nsq.py

```diff
@@ -52,36 +49,38 @@ def tearDown(self):
 
     @run_until_complete
     def test_consumer(self):
-        nsq = yield from create_nsq(host=self.host, port=self.port,
-                                    heartbeat_interval=30000,
-                                    feature_negotiation=True,
-                                    tls_v1=True,
-                                    snappy=False,
-                                    deflate=False,
-                                    deflate_level=0,
-                                    loop=self.loop)
-        for i in range(0, 100):
-            yield from nsq.pub(b'foo', b'xxx:i')
+        nsq = yield from create_writer(host=self.host, port=self.port,
+                                       heartbeat_interval=30000,
+                                       feature_negotiation=True,
+                                       tls_v1=True,
+                                       snappy=False,
+                                       deflate=False,
+                                       deflate_level=0,
+                                       loop=self.loop)
+        for i in range(100):
+            yield from nsq.pub(b'foo', f'xxx:{i}'.encode())
 
         yield from asyncio.sleep(0.1, loop=self.loop)
-        consumer = NsqConsumer(nsqd_tcp_addresses=[(self.host, self.port)],
+        reader = yield from create_reader(nsqd_tcp_addresses=[f'{self.host}:{self.port}'],
                                max_in_flight=30, loop=self.loop)
-        yield from consumer.connect()
-        yield from consumer.subscribe(b'foo', b'bar')
+        yield from reader.subscribe(b'foo', b'bar')
 
         msgs = []
-        for i, waiter in enumerate(consumer.wait_messages()):
+        for i, waiter in enumerate(reader.wait_messages()):
 
             # yield from msg.fin()
             print('-----msgs', len(msgs))
-            is_starved = consumer.is_starved()
 
-
-            if is_starved:
+            if reader.is_starved():
                 print(">>>>>>>>msgs in list: {}".format(len((msgs))))
                 for m in msgs:
                     yield from m.fin()
                 msgs = []
 
             msg = yield from waiter
             msgs.append(msg)
+
+            if i == 99:
+                break
```

Comment:
@aohan237 this test would hang up without this brake, need your advise on how to fix this.